### DISCUSSION TITLE
Camera feed

### DISCUSSION THREAD

**Discussion by amarkovits** [sceneview/sceneview#196] [Q&A]
In ARSceneView is there a way to apply shaders to the camera texture? For example to apply a shader to make it B&W or to combine with other texture for special effects?

**Comment by ThomasGorisse** [comment]
The camera feed is rendered automatically by `ARSceneView` in v4.0.0-rc.1:

```kotlin
ARSceneView(
    onSessionUpdated = { session, frame ->
        // Camera feed is displayed automatically
        // Access camera image for processing:
        val image = frame.acquireCameraImage()
        image.close()
    }
) {
    // Your AR content overlaid on the camera
}
```

If you need just the camera feed without AR (no plane detection, no anchors), consider using CameraX instead. `ARSceneView` requires ARCore and always includes AR capabilities along with the camera feed.
